src/main/java/com/synopsys/integration/alert/provider/blackduck/BlackDuckSystemValidator.java

```diff
@@ -0,0 +1,102 @@
+
+    private final AlertProperties alertProperties;
+    private final BlackDuckProperties blackDuckProperties;
+
+    @Autowired
+    public BlackDuckSystemValidator(AlertProperties alertProperties, BlackDuckProperties blackDuckProperties, SystemMessageUtility systemMessageUtility) {
+        super(systemMessageUtility);
+        this.alertProperties = alertProperties;
+        this.blackDuckProperties = blackDuckProperties;
+    }
+
+    @Override
+    public boolean validate() {
+        logger.info("Validating Black Duck Provider...");
+        getSystemMessageUtility().removeSystemMessagesByType(SystemMessageType.BLACKDUCK_PROVIDER_CONNECTIVITY);
+        getSystemMessageUtility().removeSystemMessagesByType(SystemMessageType.BLACKDUCK_PROVIDER_URL_MISSING);
+        getSystemMessageUtility().removeSystemMessagesByType(SystemMessageType.BLACKDUCK_PROVIDER_LOCALHOST);
+        try {
+            Optional<String> blackDuckUrlOptional = blackDuckProperties.getBlackDuckUrl();
+            validationCheck("Black Duck Provider Invalid; cause: Black Duck URL missing.", SystemMessageSeverity.WARNING, SystemMessageType.BLACKDUCK_PROVIDER_URL_MISSING, blackDuckUrlOptional.isEmpty())
+                .ifPresent(message -> logger.warn("  -> {}", message));
+            if (blackDuckUrlOptional.isPresent()) {
+                String blackDuckUrlString = blackDuckUrlOptional.get();
+                Boolean trustCertificate = BooleanUtils.toBoolean(alertProperties.getAlertTrustCertificate().orElse(false));
+                Integer timeout = blackDuckProperties.getBlackDuckTimeout();
+                logger.debug("  -> Black Duck Provider URL found validating: {}", blackDuckUrlString);
+                logger.debug("  -> Black Duck Provider Trust Cert: {}", trustCertificate);
+                logger.debug("  -> Black Duck Provider Timeout: {}", timeout);
+                URL blackDuckUrl = new URL(blackDuckUrlString);
+
+                validationCheck("Black Duck Provider Using localhost.", SystemMessageSeverity.WARNING, SystemMessageType.BLACKDUCK_PROVIDER_LOCALHOST, "localhost".equals(blackDuckUrl.getHost()))
+                    .ifPresent(message -> logger.warn("  -> {}", message));
+                IntLogger intLogger = new Slf4jIntLogger(logger);
+                Optional<BlackDuckServerConfig> blackDuckServerConfig = blackDuckProperties.createBlackDuckServerConfig(intLogger);
+                if (blackDuckServerConfig.isPresent()) {
+                    boolean canConnect = blackDuckServerConfig.get().canConnect(intLogger);
+                    if (canConnect) {
+                        logger.info("  -> Black Duck Provider Valid!");
+                    }
+                    validationCheck("Can not connect to the Black Duck server with the current configuration.", SystemMessageSeverity.WARNING, SystemMessageType.BLACKDUCK_PROVIDER_CONNECTIVITY, !canConnect)
```

Comment:
This is minor, but "can not" is typically combined into "cannot". I only mention this to be consistent with other error messages in the application. Either form is still grammatically correct.